exp/api/v1alpha3/azuremanagedmachinepool_types.go

```diff
@@ -23,6 +23,10 @@ import (
 
 // AzureManagedMachinePoolSpec defines the desired state of AzureManagedMachinePool.
 type AzureManagedMachinePoolSpec struct {
+	// Mode - represents mode of an agent pool. Possible values include: System, User.
+	// +kubebuilder:validation:Enum=System;User
+	Mode string `json:"mode"`
```

Comment:
should we default this? or require it? i'm fine as-is but thought I'd raise it

Reply:
I am okay with anything here I think. I guess the only difference would be that the user gets more knowledgeable about the mode of the machinepool if kept as required 😅.

Reply:
I am okay with anything

Reply:
Have we decided on this ? Should I add a default instead of required ?

Reply:
Changing from required => optional is non-breaking, let’s leave it required for now 👍